labelbox/client.py

```diff
@@ -939,3 +940,306 @@ def get_model_run(self, model_run_id: str) -> ModelRun:
+            assignGlobalKeysToDataRows(data: {assignInputs: $globalKeyDataRowLinks}) { 
+                jobId 
+            }
+        }
+        """
+        params = {
+            'globalKeyDataRowLinks': [{
+                utils.camel_case(key): value for key, value in input.items()
+            } for input in global_key_to_data_row_inputs]
+        }
+        assign_global_keys_to_data_rows_job = self.execute(query_str, params)
+
+        # Query string for retrieving job status and result, if job is done
+        result_query_str = """query assignGlobalKeysToDataRowsResultPyApi($jobId: ID!) {
+            assignGlobalKeysToDataRowsResult(jobId: {id: $jobId}) {
+                jobStatus
+                data {
+                    sanitizedAssignments {
+                        dataRowId
+                        globalKey
+                    }
+                    invalidGlobalKeyAssignments {
+                        dataRowId
+                        globalKey
+                    }
+                    unmodifiedAssignments {
+                        dataRowId
+                        globalKey
+                    }
+                    accessDeniedAssignments {
+                        dataRowId
+                        globalKey
+                    }
+                }}}
+        """
+        result_params = {
+            "jobId":
+                assign_global_keys_to_data_rows_job["assignGlobalKeysToDataRows"
+                                                   ]["jobId"]
+        }
+
+        # Poll job status until finished, then retrieve results
+        sleep_time = 2
+        start_time = time.time()
+        while True:
+            res = self.execute(result_query_str, result_params)
+            if res["assignGlobalKeysToDataRowsResult"][
+                    "jobStatus"] == "COMPLETE":
+                results, errors = [], []
+                res = res['assignGlobalKeysToDataRowsResult']['data']
+                # Successful assignments
+                results.extend(
+                    _format_successful_assignments(
+                        assignments=res['sanitizedAssignments'],
+                        sanitized=True))
+                results.extend(
+                    _format_successful_assignments(
+                        assignments=res['unmodifiedAssignments'],
+                        sanitized=False))
+                # Failed assignments
+                errors.extend(
+                    _format_failed_assignments(
+                        assignments=res['invalidGlobalKeyAssignments'],
+                        error_msg="Invalid global key"))
+                errors.extend(
+                    _format_failed_assignments(
+                        assignments=res['accessDeniedAssignments'],
+                        error_msg="Access denied to Data Row"))
+
+                if len(errors) == 0:
```

Comment:
**suggestion** if errors is an empty list, we can rewrite this as


```suggestion
if not errors:
    status = "Success"
elif errors and results:
    status = "Partial Success"
else:
    status = "Failure"
                
                
```

Reply:
**nit** also, we have used enums to reflect possible paths instead of just text. might be better to have that? 

though i am unsure if it falls under the same category as the previous `input` where it was unnecessary, but users ideally would not need to do any additional importing/have to read it. and instead, it is possible to extend it if we wanted to continue on with messaging for success/partial/failure

```
Class SuccessType(enum):

SUCCESS = auto()
PARTIAL = auto()
FAILURE = auto()
```